Source/WebCore/platform/graphics/avfoundation/objc/MediaSampleAVFObjC.mm

```diff
@@ -288,6 +288,22 @@ static bool isCMSampleBufferNonDisplaying(CMSampleBufferRef sample)
     return true;
 }
 
+Vector<Ref<MediaSampleAVFObjC>> MediaSampleAVFObjC::divide()
+{
+    auto numSamples = PAL::CMSampleBufferGetNumSamples(m_sample.get());
+
+    if (numSamples == 1)
+        return Vector<Ref<MediaSampleAVFObjC>>::from(Ref { *this });
```

Comment:
It is a bit strange to `divide` even in case where `isDivisable` is false.
I guess this is more convenient?

Reply:
yes, it returns itself so that you can do `if (auto& sample : samples.divide())`. this is what `divideIntoHomogeneousSamples` does currently.